### ISSUE TITLE
Do not announce field as invalid before it is shown as such

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#4797]
We put `aria-invalid="true"` on the `input`, and this is done in sync with the `xforms-invalid` class on the control, but in Form Runner we only show the error when the field also gets `xforms-visited`, this at the Form Runner level, with CSS. The problem is that as far as I know we can't have equivalent CSS that tells the screen reader "don't read the `aria-invalid="true"` until the parents gets a `xforms-visited`. (For that we would need CSS selectors to be able to target attributes, not just to target elements based on the value of attributes.)

[+1 from customer](https://3.basecamp.com/3600924/buckets/2016476/messages/3528969610)

**Comment by avernet** [comment]
The simplest fix would be for the XForms engine to only put `aria-invalid="true"` when it puts both the `xforms-invalid` and `xforms-visited` classes.

To be more consistent and backward compatible for those who use XForms without Form Runner, we might want to make this configurable, and have Form Runner tell XForms to do this, by default keeping the current behavior. But since this adds complexity, which might not be warranted for, we might only want to do it later if needed.
